### ISSUE TITLE
rocketeer flow not working

### ISSUE THREAD

**Issue by gophry** [rocketeers/rocketeer#641]
Assuming the flow is to first perform:
```shell
rocketeer check
rocketeer setup
rocketeer deploy
```
- [ ] The Composer package manager could not be found

1. $ [ -e "/home/www/laravel/current/composer.json" ] && echo "true"
for a fresh deployment there would be now composer.json present. 
[rocketeer check - composer.txt](https://github.com/rocketeers/rocketeer/files/142491/rocketeer.check.-.composer.txt)
2. creating a current/composer.json we can proceed.

- [ ] cannot create directory [rocketeer setup]
[root@A.B.C.D] (production) mkdir:
[root@A.B.C.D] (production) cannot create directory ‘/var/www/laravel/’
[root@A.B.C.D] (production) : File exists
[root@A.B.C.D] (production) 
[rocketeer-setup.txt](https://github.com/rocketeers/rocketeer/files/142559/rocketeer-setup.txt)

if rocketeer deploy is performed for the first time with all required drivers installed then all works fine.

**Comment by aik099** [comment]
> if rocketeer deploy is performed for the first time with all required drivers installed then all works fine.

What you mean by drivers in relation to Composer?

You should already have `composer` binary present on server to make deploying possible.

**Comment by gophry** [comment]
Yes composer is present. But option check also checks for composer.json present in current folder which will be created only after deploy.

this is in regards to a fresh deployment.

**Comment by aik099** [comment]
I did exactly as you, but without running `setup` command. Maybe `composer` isn't in your PATH? In my case I've put it into `/usr/local/bin` folder which is in $PATH variable when executing commands over SSH (I've connect to server over ssh).

**Comment by gophry** [comment]
It is surely in path and is also detected by rocketeer. the issue is not with path. its with composer.json file check surely. if a composer.json exists in the project path then only check shows completed.
PATH I have tried all possible mentioned in $PATH even added a custom path and checked.

rocketeer version 2.1.2

**Comment by aik099** [comment]
Can you please post complete deployment log from `rocketeer.phar deploy -v` command (with private information hidden) as Gist?

Or if your project is public, then just share its repo url so that I can deploy it locally and see what happens (I presume I will see the same error).

In my case `Checking presence of required drivers` step passes and then cloning happens, which creates `composer.json` (because it's in the repo) and then composer related step is executed.

**Comment by gophry** [comment]
attached files already.
please check this.
https://github.com/rocketeers/rocketeer/files/142491/rocketeer.check.-.composer.txt

**Comment by aik099** [comment]
> But i will be always checking if the server is ready for deployment and then deploy right?

I don't know.

After reading a log I see `The Composer package manager could not be found` message and from log context I now understand where it happens. The same check for me passes. Here are the ideas what can be wrong:

* the `which php` command fails
* the `which composer` command fails

Am I right?

**Comment by gophry** [comment]
Yes first check and then deploy.

The which php and which composer does not fail.

There is check for composer.json on remote server which fails. Mentioned
the line check first entry in issue.

**Comment by aik099** [comment]
The `AbstractPackageManager::hasManifest` method, that checks for existence of `composer.json` file works like this:

1. look if file exists in current release
2. if it doesn't look if file exists in the directory you're running Rocketeer from

This works in my case, because I'm running `rocketeer.phar check` command right in cloned repo which I have on development machine, which has `.rocketeer` folder and `composer.json` file on same level.

Have you exactly same setup?

**Comment by gophry** [comment]
No not the same setup rocketeer is in usr/bin.

**Comment by aik099** [comment]
So full path to rocketeer is `/usr/bin/rocketeer.phar` or just `/usr/bin/rocketeer`. And when you run the command you're:

* in the project folder that will be deployed
* that project has `.rocketeer` sub-folder

?

**Comment by aik099** [comment]
I'm using Rocketeer 2.1.2 and above workflow (when I have Rocketeer placed centrally on developer machine) also works without problems.

Do you really have this folder structure in the folder you're running `rocketeer check` command from:

```
.rocketeer/
...
composer.json
```

?

**Comment by gophry** [comment]
Not exactly. Scenario is like this.
I am on a central deploy server and installed composer, rocketeer, PHP7.0
there is a user deploy, from the home folder: rocketeer check

The project does not reside on this central server.

Final Scenario:
There is a small app(webhook) which recieves push from  github on which the
rocketeer command will be executed as user deploy.

**Comment by gophry** [comment]
Also,

.rocketeer folder strucuture is present in the /home/deploy user folder.

**Comment by aik099** [comment]
I'm not sure, that this scenario was even supposed to work based on what I'm reading in docs. In fact does doesn't explicitly say where Rocketeer commands are supposed to be executed and I've just presumed that it should happen in a clone of a repo that will be deployed.

At least now you know how to make it work: in the home folder of `deploy` user do a clone of repo that will be deployed.
